src/cfml/system/util/CompileDSL.cfc

```diff
@@ -0,0 +1,402 @@
+					type="commandException"
+				);
+
+			}
+			return currentPath;
+
+		} ) );
+
+        job.addLog( " " & serialize( getSourcePaths() ) & " " );
+		try{
+
+			writeTempSourceFile( tempSrcFileName );
+
+			var javacCommand = 'run "#getJavaBinFolder()#javac" "@#tempSrcFileName#" -d #variables.classOutputDirectory# #variables.compileOptionsString#';
+
+			/* if ( getVerbose() ) {
+				javacCommand &= " -verbose";
+			} */
+
+			if ( getEncode().len() ){
+				javacCommand &= " -encoding #variables.encode#";
+			}
+
+			//shell.printString( " " & javacCommand & " " );
+            job.addLog( " " & javacCommand & " " );
+			command( javacCommand ).run();
+
+		} finally {
+			if ( FileExists( tempSrcFileName ) ){
+				fileDelete( tempSrcFileName );
+			}
+
+		}
+
+	}
+
+	function writeTempSourceFile( string tempSrcFileName , array sourcePath=getSourcePaths() , string extension=".java" ) {
+		var globber = wirebox.getInstance( 'globber' );
+
+		//shell.printString( " gSP-> #serialize(getSourcePaths())# " );
+        job.addLog( " gSP-> #serialize(sourcePath)# " );
+		var sourceList = globber
+				.setPattern( sourcePath )
+				.asQuery()
+				.matches()
+				.filter(( row ) => row.type=="file" && row.name.endsWith( extension ))
+				.reduce(( acc, row ) => {
+					return listappend( acc, row.directory & "/" & row.name, chr(10) );
+				}, "")
+
+        job.addLog( " sList-> #serialize(sourceList)# " );
+		if( !sourceList.len() ) {
+			/* throw(
+				message='No java source files found in [#sourceDirs.toList()#]', detail='Check fromSource() and try again',
+				type="commandException"
+			); */
+			throw(
+				message='No java source files found in [#sourceList.len()#]', detail='Check fromSource() and try again',
+				type="commandException"
+			);
+		}
+
+		fileWrite( tempSrcFileName, sourceList );
+	}
+
+	function buildJar() {
+		var currentLibsDir = fileSystemutil.resolvePath( getLibsDir(), getProjectRoot() );
+		var jarName = getJarNameString();
+        var currentProjectRoot = getProjectRoot();
+
+        var tempSrcFileName = tempDir & 'temp#createUUID()#.txt';
+
+		//var currentLibsDir = fileSystemutil.resolvePath( getLibsDir(), getProjectRoot() );
+		//shell.printString( ' cLD-> ' & currentLibsDir );
+
+        // if the jarname is empty
+        // we check if the current libs is a package
+        // if its a package
+        // get name from package
+        // if not get name from folder
+        // if its root
+        // give default name
+
+        job.start( ' for build jar check jarName ' );
+        if( jarName.len() == 0 ){
+            // jarName is empty
+            //shell.printString( ' jarName is empty ' );
+            job.addLog( ' jarName is empty ' );
+            job.addLog( ' currentLibsDir-> #currentLibsDir# ' );
+            job.addLog( ' currentProjectRoot-> #currentProjectRoot# ' );
+            if( packageService.isPackage( currentProjectRoot ) ) {
```

Comment:
I would move this check inside the `getJarnameFromPackage()` method and simply return an empty string if the method isn't able to find a name for the jar.  (See my comments inside that method).  Then if the name is still empty, you can continue on to the next try.

Reply:
done